### PR TITLE
feat: Engagement Center Add permalink to the engagement center tabs - MEEDS-449 - meeds-io/meeds#13

### PR DESCRIPTION
**Pull request by ArfaouiMohamedAmine**
This change is going to add `permalinks` to the `engagement center tabs` in order to be accessed easily using the followings;
`{DOMAIN}/portal/{SITE}/engagement/challenges`
`{DOMAIN}/portal/{SITE}/engagement/programs`
`{DOMAIN}/portal/{SITE}/engagement/achievement`

### CODE REVIEW

**Review by AzmiTouil — DISMISSED** [review]

File: challenges-webapp/src/main/resources/locale/navigation/portal/global_en.properties [hunk 1]
```diff
@@ -15,3 +15,5 @@
 # Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 #
 portal.global.challenges=Challenges
+portal.global.engagement=Engagement
```
Comment (plamarque) [inline challenges-webapp/src/main/resources/locale/navigation/portal/global_en.properties]:
```suggestion
portal.global.contributions=Contributions
```

File: challenges-webapp/src/main/webapp/WEB-INF/conf/challenges/app-center-configuration.xml [hunk 2]
```diff
@@ -59,7 +59,7 @@ Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
                             <string>Challenges</string>
                         </field>
                         <field name="url">
-                            <string>./challenges</string>
+                            <string>./engagement</string>
```
Comment (plamarque) [inline challenges-webapp/src/main/webapp/WEB-INF/conf/challenges/app-center-configuration.xml]:
```suggestion
                            <string>./contributions</string>
```

File: challenges-webapp/src/main/webapp/WEB-INF/conf/challenges/portal/portal/global/navigation.xml [hunk 3]
```diff
@@ -30,5 +30,11 @@ Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
             <visibility>SYSTEM</visibility>
             <page-reference>portal::global::challenges</page-reference>
         </node>
+        <node>
+            <name>engagement</name>
```
Comment (plamarque) [inline challenges-webapp/src/main/webapp/WEB-INF/conf/challenges/portal/portal/global/navigation.xml]:
```suggestion
            <name>contributions</name>
```

File: challenges-webapp/src/main/webapp/WEB-INF/conf/challenges/portal/portal/global/navigation.xml [hunk 4]
```diff
@@ -30,5 +30,11 @@ Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
...
+            <label>#{portal.global.engagement}</label>
```
Comment (plamarque) [inline challenges-webapp/src/main/webapp/WEB-INF/conf/challenges/portal/portal/global/navigation.xml]:
```suggestion
            <label>#{portal.global.contributions}</label>
```

File: challenges-webapp/src/main/webapp/WEB-INF/conf/challenges/portal/portal/global/navigation.xml [hunk 5]
```diff
@@ -30,5 +30,11 @@ Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
...
+            <visibility>SYSTEM</visibility>
+            <page-reference>portal::global::engagement</page-reference>
```
Comment (plamarque) [inline challenges-webapp/src/main/webapp/WEB-INF/conf/challenges/portal/portal/global/navigation.xml]:
```suggestion
            <page-reference>portal::global::contributions</page-reference>
```

File: challenges-webapp/src/main/webapp/WEB-INF/conf/challenges/portal/portal/global/pages.xml [hunk 6]
```diff
@@ -21,8 +21,8 @@ Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
         xsi:schemaLocation="http://www.gatein.org/xml/ns/gatein_objects_1_6 http://www.gatein.org/xml/ns/gatein_objects_1_6"
         xmlns="http://www.gatein.org/xml/ns/gatein_objects_1_6">
   <page>
-    <name>challenges</name>
-    <title>challenges</title>
+    <name>engagement</name>
```
Comment (plamarque) [inline challenges-webapp/src/main/webapp/WEB-INF/conf/challenges/portal/portal/global/pages.xml]:
```suggestion
    <name>contributions</name>
```

File: challenges-webapp/src/main/webapp/WEB-INF/conf/challenges/portal/portal/global/pages.xml [hunk 7]
```diff
@@ -21,8 +21,8 @@ Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
...
+    <title>engagement</title>
```
Comment (plamarque) [inline challenges-webapp/src/main/webapp/WEB-INF/conf/challenges/portal/portal/global/pages.xml]:
```suggestion
    <title>contributions</title>
```

File: challenges-webapp/src/main/webapp/vue-app/engagement-center/components/challenges/ChallengeDetailsDrawer.vue [hunk 8]
```diff
@@ -179,7 +179,7 @@ export default {
   methods: {
     open(challenge) {
       this.challenge = challenge;
-      window.history.replaceState('challenges', this.$t('challenges.challenges'), `${eXo.env.portal.context}/${eXo.env.portal.portalName}/challenges/${this.challenge.id}`);
+      window.history.replaceState('challenges', this.$t('challenges.challenges'), `${eXo.env.portal.context}/${eXo.env.portal.portalName}/engagement/challenges/${this.challenge.id}`);
```
Comment (plamarque) [inline challenges-webapp/src/main/webapp/vue-app/engagement-center/components/challenges/ChallengeDetailsDrawer.vue]:
```suggestion
      window.history.replaceState('challenges', this.$t('challenges.challenges'), `${eXo.env.portal.context}/${eXo.env.portal.portalName}/contributions/challenges/${this.challenge.id}`);
```

File: challenges-webapp/src/main/webapp/vue-app/engagement-center/components/challenges/ChallengeDetailsDrawer.vue [hunk 9]
```diff
@@ -195,7 +195,7 @@ export default {
         }).finally(() => this.$refs.challengeDetails.endLoading());
     },
     close() {
-      window.history.replaceState('challenges', this.$t('challenges.challenges'), `${eXo.env.portal.context}/${eXo.env.portal.portalName}/challenges`);
+      window.history.replaceState('challenges', this.$t('challenges.challenges'), `${eXo.env.portal.context}/${eXo.env.portal.portalName}/engagement/challenges`);
```
Comment (plamarque) [inline challenges-webapp/src/main/webapp/vue-app/engagement-center/components/challenges/ChallengeDetailsDrawer.vue]:
```suggestion
      window.history.replaceState('challenges', this.$t('challenges.challenges'), `${eXo.env.portal.context}/${eXo.env.portal.portalName}/contributions/challenges`);
```

**Review by plamarque — COMMENTED** [review]
aSuggestion to adopt the term "Contributions" (Contributions Center, contribution program, etc.) instead of Engagement.

CC @Samuel-Renault-eXo

**Comment by srenault-meeds** [comment]
> 

Let's stick to Engagement for now and we will propose one task to review the labelling once it is validated.

**Comment by AzmiTouil** [comment]
> > 
> 
> Let's stick to Engagement for now and we will propose one task to review the labelling once it is validated.

I think we need to work on it in the same task, because it's not just the labeling, it's also changing the page name and the navigation name

**Comment by ArfaouiMohamedAmine** [comment]
@AzmiTouil ready for review

**Review by AzmiTouil — APPROVED** [review]

**Pull request merged**